components/plan-build/bin/environment.sh

```diff
@@ -238,6 +238,54 @@ __populate_environment_from_deps() {
     fi
 }
 
+# For dependencies that do not have a RUNTIME_ENVIRONMENT file, we'll
+# at least assemble a PATH variable that can be used. To be complete
+# and utterly paranoid, we'll iterate through the TDEPS of the
+# dependency and assemble a PATH by layering all the PATHs of each of
+# those dependencies.
+#
+# Don't need to do this for packages that already have a
+# RUNTIME_ENVIRONMENT file, because this operation will have already
+# been done when those packages were built. This is to help preserve
+# backward compatibility.
+__assemble_legacy_path() {
+    local path_to_dep="${1}"
+    local path=""
+
+    warn "Assembling legacy path for ${path_to_dep}"
+    local tdeps=$(_read_metadata_file_for ${path_to_dep} "TDEPS")
```

Comment:
I've learned that if you split the variable declaration and variable setting that you'll be able to propagate an error back on `tdeps=$(_read_metadata_file_for ${path_to_dep} "TDEPS")` if `_read_metadata_file_for()` fails, whereas `local tdeps=$(_read_metadata_file_for ${path_to_dep} "TDEPS")` always returns 0.

I know we have more than a few places with this issue--I only learned later about this and it's helped in some places.

Reply:
Oh, that's nice to know! I'll make those tweaks